cms/src/main/java/uk/nhs/digital/apispecs/swagger/CodegenDefaultGenerator.java

```diff
@@ -0,0 +1,194 @@
+                if (authMethods == null || authMethods.isEmpty()) {
+                    authMethods = getAuthMethods(globalSecurities, securitySchemes);
+                }
+
+                if (authMethods != null && !authMethods.isEmpty()) {
+                    codegenOperation.authMethods = config.fromSecurity(authMethods);
+                    codegenOperation.getVendorExtensions().put(CodegenConstants.HAS_AUTH_METHODS_EXT_NAME, Boolean.TRUE);
+                }
+            } catch (Exception ex) {
+                ex.printStackTrace();
+                String msg = "Could not process operation:\n" //
+                    + "  Tag: " + tag + "\n"//
+                    + "  Operation: " + operation.getOperationId() + "\n" //
+                    + "  Resource: " + httpMethod + " " + resourcePath + "\n"//
+                    // + "  Definitions: " + swagger.getDefinitions() + "\n"  //
+                    + "  Exception: " + ex.getMessage();
+                throw new RuntimeException(msg, ex);
+            }
+        }
+
+    }
+
+    private static String generateParameterId(Parameter parameter) {
+        return parameter.getName() + ":" + parameter.getIn();
+    }
+
+    private Map<String, SecurityScheme> getAuthMethods(List<SecurityRequirement> securities, Map<String, SecurityScheme> securitySchemes) {
+        if (securities == null || securitySchemes == null || securitySchemes.isEmpty()) {
+            return null;
+        }
+        final Map<String, SecurityScheme> authMethods = new HashMap<>();
+        for (SecurityRequirement requirement : securities) {
+            for (String key : requirement.keySet()) {
+                SecurityScheme securityScheme = securitySchemes.get(key);
+                if (securityScheme != null) {
+                    authMethods.put(key, securityScheme);
+                }
+            }
+        }
+        return authMethods;
+    }
+
+    private Map<String, List<CodegenOperation>> orderedByDeclaredTags(final Map<String, List<CodegenOperation>> operationsByTags) {
+        final List<String> declaredTagNames = operationsByTags.containsKey("Default")
+            ? new ArrayList<>(singletonList("Default"))
+            : new ArrayList<>(operationsByTags.size());
+
+        declaredTagNames.addAll(
+            Optional.ofNullable(openAPI.getTags())
+                .orElse(Collections.emptyList())
+                .stream()
+                .map(Tag::getName)
+                .map(config::sanitizeTag)
+                .collect(toList())
+        );
+
+        final Map<String, List<CodegenOperation>> orderedOperationsByTags = new LinkedHashMap<>(operationsByTags.size());
+
+        declaredTagNames.stream()
+            .filter(operationsByTags::containsKey)
+            .forEachOrdered(tagName -> orderedOperationsByTags.put(tagName, operationsByTags.get(tagName)));
+
+        return orderedOperationsByTags;
+    }
+
+    @Override
+    public List<File> generate() {
```

Comment:
I know this override was included in my example patch but I believe it can actually go.

The patch was generated before the fix implemented by this override was replaced by `ApiSpecificationStaticHtml2Codegen#populateComponentsFieldWithEmptyObjectWhenNull`.